admin/cluster_setup.py

```diff
@@ -0,0 +1,278 @@
+                print("Be sure to preserve the required files.")
+
+    raise SystemExit(result)
+
+
+def _build_config(cluster, application_template, per_node):
+    """
+    Build a Flocker deployment configuration for the given cluster
+    and parameters.
+    The configuration consists of identically configured applications
+    (containers) uniformly spread over all cluster nodes.
+
+    :param flocker.provision._common.Cluster cluster: The target cluster.
+    :param dict application_template: A dictionary that provides configuration
+                                      for an individual application.
+    :param int per_node: The number of applications to deploy on each cluster
+                         node.
+    :return dict: The deployment configuration.
+    """
+    application_root = {}
+    applications = {}
+    application_root["version"] = 1
+    application_root["applications"] = applications
+    for node in cluster.agent_nodes:
+        for i in range(per_node):
+            name = "app_%s_%d" % (node.private_address, i)
+            applications[name] = deepcopy(application_template)
+
+    deployment_root = {}
+    nodes = {}
+    deployment_root["nodes"] = nodes
+    deployment_root["version"] = 1
+    for node in cluster.agent_nodes:
+        nodes[node.private_address] = []
+        for i in range(per_node):
+            name = "app_%s_%d" % (node.private_address, i)
+            nodes[node.private_address].append(name)
+
+    return {"applications": application_root,
+            "deployment": deployment_root}
+
+
+def _configure(reactor, cluster, configuration):
+    """
+    Configure the cluster with the given deployment configuration.
+
+    :param reactor: The reactor to use.
+    :param flocker.provision._common.Cluster cluster: The target cluster.
+    :param dict configuration: The deployment configuration.
+    :return Deferred: Deferred that fires when the configuration is pushed
+                      to the cluster's control agent.
+    """
+    base_url = b"https://{}:{}/v1".format(
+        cluster.control_node.address, REST_API_PORT
+    )
+    certificates_path = cluster.certificates_path
+    cluster_cert = certificates_path.child(b"cluster.crt")
+    user_cert = certificates_path.child(b"user.crt")
+    user_key = certificates_path.child(b"user.key")
+    body = dumps(configuration)
+    treq_client = treq_with_authentication(
+        reactor, cluster_cert, user_cert, user_key)
+
+    def got_all_nodes():
+        d = treq_client.get(
+            base_url + b"/state/nodes",
+            persistent=False
+        )
+        d.addCallback(check_and_decode_json, OK)
+        d.addCallback(
+            lambda nodes: len(nodes) >= len(cluster.agent_nodes)
+        )
+        d.addErrback(write_failure, logger=None)
+        return d
+
+    got_nodes = loop_until(reactor, got_all_nodes, repeat(1, 300))
+
+    def do_configure(success):
+        if not success:
+            return 1
```

Comment:
Rather than passing a non-zero return value through the `do_configure` function, raise exceptions as necessary.

Unconditionally set `result = 0` inside the `try...except` in `main`, and rely on the `except` clause to set result to 1.